Flow.Launcher/ViewModel/MainViewModel.cs

```diff
@@ -780,10 +738,169 @@ public string VerifyOrSetDefaultHotkey(string hotkey, string defaultHotkey)
+            }
+
+        }
 
-        public bool PreviewVisible { get; set; } = false;
+        private void HidePreview()
+        {            
+            if (PluginManager.UseExternalPreview())
+                CloseExternalPreview();
+
+            if (InternalPreviewVisible)
+                HideInternalPreview();
+        }
+
+        [RelayCommand]
+        private void TogglePreview()
+        {
+            switch (InternalPreviewVisible || ExternalPreviewVisible)
+            {
+                case true:
+                    HidePreview();
+                    break;
+
+                case false:
+                    ShowPreview();
+                    break;
+            }
+        }
+
+        private void ToggleInternalPreview()
+        {
+            if (!InternalPreviewVisible)
+            {
+                ShowInternalPreview();
+            }
+            else
+            {
+                HideInternalPreview();
+            }
+        }
+
+        private void OpenExternalPreview(string path, bool sendFailToast = true)
+        {
+            _ = PluginManager.OpenExternalPreviewAsync(path, sendFailToast).ConfigureAwait(false);
+            ExternalPreviewVisible = true;
+        }
+
+        private void CloseExternalPreview()
+        {
+            _ = PluginManager.CloseExternalPreviewAsync().ConfigureAwait(false);
+            ExternalPreviewVisible = false;
+        }
+
+        private void SwitchExternalPreview(string path, bool sendFailToast = true)
+        {
+            _ = PluginManager.SwitchExternalPreviewAsync(path,sendFailToast).ConfigureAwait(false);
+        }
 
-        public int ResultAreaColumn { get; set; } = 1;
+        private void ShowInternalPreview()
+        {
+            ResultAreaColumn = ResultAreaColumnPreviewShown;
+            Results.SelectedItem?.LoadPreviewImage();
+        }
+
+        private void HideInternalPreview()
+        {
+            ResultAreaColumn = ResultAreaColumnPreviewHidden;
+        }
+
+        public void ResetPreview()
+        {
+            switch (Settings.AlwaysPreview)
+            {
+                case true:
+                    ShowPreview();
+                    break;
+
+                case false:
+                    HidePreview();
+                    break;
+            }
+        }
+        
+        private void UpdatePreview()
+        {
+            if (InternalPreviewVisible)
+            {
+                Results.SelectedItem?.LoadPreviewImage();
+                return;
+            }
+
+            switch (PluginManager.UseExternalPreview())
+            {
+                case true
+                    when ExternalPreviewVisible && CanExternalPreviewSelectedResult(out var path):
+                    SwitchExternalPreview(path, false);
+                    break;
+
+                case true
+                    when !ExternalPreviewVisible && Settings.AlwaysPreview && CanExternalPreviewSelectedResult(out var path):
```

Comment:
When I enable AlwaysPreview QuickLook automatically pops whenever I type. Not usable imo.

Reply:
Yeah i know, there is the option to switch AlwaysPreview off. The real problem is QuickLook doesnt remember last position, but nothing we can do.

Reply:
So AlwaysPreview shouldn't control external preview.

Reply:
I want to align the behaviour for both internal and external so logic is a bit more straightforward, and also not just build logic for QuickLook as in the future we may integrate with another preview tool that works well with always preview option. For now user can just switch it off.

Reply:
Is it possible to set a "always external preview" in preview plugin?

Reply:
Hmm, let's not not add additional logic when there is already a button that does exactly that.